be/src/exec/pipeline/query_context.h

```diff
@@ -198,6 +219,18 @@ class QueryContext : public std::enable_shared_from_this<QueryContext> {
 
     int64_t get_static_query_mem_limit() const { return _static_query_mem_limit; }
 
+    void set_max_concurrency(size_t max_concurrency) { _max_driver_concurrency = max_concurrency; }
+
+    ConcurrencyToken acquire_exec_concurrency() {
+        if (_driver_concurrency.fetch_add(1) == _max_driver_concurrency - 1) {
```

Comment:
Maybe `_driver_concurrency.fetch_add(1) >= _max_driver_concurrency - 1)`? Maybe multiple drivers call `acquire_exec_concurrency()` at the same time, resulting that `_driver_concurrency.fetch_add(1)` may be greater than `_max_driver_concurrency - 1`.

Reply:
Will this happen?
Assume that `_max_driver_concurrency=1`, `_driver_concurrency = 0`.
1. T1 calls `_driver_concurrency.fetch_add(1)`, so `_driver_concurrency1` is 1 after this.
2. Before T1 calling `_driver_concurrency--`, t2 calls `_driver_concurrency.fetch_add(1)`, then `_driver_concurrency1` will returns 1 which is greater than `_max_driver_concurrency-1`.